src/validationinterface.cpp

```diff
@@ -126,52 +131,90 @@ void SyncWithValidationInterfaceQueue() {
     promise.get_future().wait();
 }
 
+// Use a macro instead of a function for conditional logging to prevent
```

Comment:
Not a C++ expert, could you explain or point to why macro prevent to evaluate argument there? I mean aren't they replaced at preprocessing and we know flag only at runtime?

Reply:
compare the macro expansion:

```c++
if (debug_foo) {
    LogPrintf("Foobar %i", ExpensiveToComputeFunction());
}
```

with

```c++
LogPrint(BCLog::FOO, "Foobar %i", ExpensiveToComputeFunction());
```

In the first example, `ExpensiveToComputeFunction` is not invoked if `debug_foo` is false. In the second one it's invoked, but then disregarded by `LogPrint`.

Reply:
Thanks but got it, `LogPrint` has been changed to a macro in https://github.com/bitcoin/bitcoin/pull/17218, was still thinking it was a function...

That's said shouldn't we use the `LOG_PRINT` convention as for any other macro ?

Reply:
I'm using the convention in this PR but did not change it in the referenced PR for `LogPrint` since it was simply partially reverting a commit. It would have been a much larger change to update all the call sites, and that's something I'd be hesitant in undertaking without enough support for the change.